src/Processors/QueryPlan/Optimizations/joinRuntimeFilter.cpp

```diff
@@ -74,6 +74,10 @@ bool tryAddJoinRuntimeFilter(QueryPlan::Node & node, QueryPlan::Nodes & nodes, c
     if (node.children.size() != 2)
         return false;
 
+    /// If right table is already prepared then runtime filter cannot be constructed
+    if (typeid_cast<JoinStepLogicalLookup *>(node.children[1]->step.get()))
```

Comment:
What does `prepared` mean in this context? There is no any documentation in the `JoinStepLogicalLookup` code, so it's unclear what does this `if` actually mean